### PR TITLE
Do not duplicate the code for deployer setup

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: playbooks/roles/airship-setup-deployer/tasks/kubectl-install.yml [hunk 1]
```diff
@@ -1,96 +0,0 @@
...
-    - name: Find if runner has a kubeconfig file
-      stat:
-        path: "{{ kubeconfig_file_path }}"
-      delegate_to: localhost
-      register: new_config
-
-    - name: Find if an existing .kube/config file
-      stat:
-        path: "{{ ansible_user_dir }}/.kube/config"
-      delegate_to: localhost
```
Comment (jhesketh) [inline playbooks/roles/airship-setup-deployer/tasks/kubectl-install.yml]: playbooks/roles/deploy-osh/tasks/kubectl-install.yml does not have this delegation.

File: playbooks/roles/airship-setup-deployer/tasks/main.yml [hunk 2]
```diff
@@ -43,17 +43,23 @@
     append: yes
 
 - name: Ensure kubectl is installed
-  include: kubectl-install.yml
+  include_role:
```
Comment (jgu17) [inline playbooks/roles/airship-setup-deployer/tasks/main.yml]: I like the idea here. But the airship roles are the ones used for the product deployment. I'd recommend go the other direction to have deploy_osh to include the airship role instead.

File: playbooks/roles/airship-setup-deployer/tasks/main.yml [hunk 3]
```diff
@@ -43,17 +43,23 @@
...
+    name: deploy_osh
+    tasks_from: kubectl-install
   tags:
     - install
 
 - name: Ensure helm and tiller are up to date
-  include: helm-install.yml
+  include_role:
```
Comment (jgu17) [inline playbooks/roles/airship-setup-deployer/tasks/main.yml]: same comment as for line 46

File: playbooks/roles/airship-setup-deployer/tasks/main.yml [hunk 4]
```diff
@@ -43,17 +43,23 @@
...
+    name: deploy_osh
+    tasks_from: helm-install
   tags:
     - install
 
 - name: Ensure helm runs on localhost
-  include: helm-run.yml
+  include_role:
```
Comment (jgu17) [inline playbooks/roles/airship-setup-deployer/tasks/main.yml]: same comments for line 46

File: playbooks/roles/airship-setup-deployer/tasks/main.yml [hunk 5]
```diff
@@ -86,17 +92,9 @@
...
-    - /etc/openstack
-
-- name: Copy clouds.yml
-  become: yes
+- name: Save default clouds.yaml file
   template:
-    src: clouds.yml.j2
-    dest: /etc/openstack/clouds.yaml
+    src: clouds.yaml.j2
+    dest: /root/.config/openstack/clouds.yaml
```
Comment (jgu17) [inline playbooks/roles/airship-setup-deployer/tasks/main.yml]: why moving to /root?

File: playbooks/roles/airship-setup-deployer/tasks/main.yml (hunk 2 above)
Comment (jsuchome) [reply playbooks/roles/airship-setup-deployer/tasks/main.yml]: In general I would not care, but there's slight difference in one of the file where we've already added some change in osh role but not in airship

File: playbooks/roles/airship-setup-deployer/tasks/main.yml (hunk 5 above)
Comment (jsuchome) [reply playbooks/roles/airship-setup-deployer/tasks/main.yml]: I'm actually not sure now why did I do that change originally (there's a link to the original commit).  I remember @dirkmueller suggested it and it sounded reasonable

File: playbooks/roles/airship-setup-deployer/tasks/main.yml (hunk 2 above)
Comment (jgu17) [reply playbooks/roles/airship-setup-deployer/tasks/main.yml]: I don't care which version to keep either. If the osh version has more features than the airship version, we can move the osh version over to the airship role folder.

File: playbooks/roles/airship-setup-deployer/tasks/main.yml (hunk 5 above)
Comment (jgu17) [reply playbooks/roles/airship-setup-deployer/tasks/main.yml]: does that mean now we require only run as root user?

Comment (jsuchome) [reply playbooks/roles/airship-setup-deployer/tasks/main.yml]: That would require it, yes. Actually you might be right that keeping it under /etc/ looks better

**Comment by jsuchome** [comment]
@jgu17 I removed the cloud config file change. Still keeping OSH files as the only one, as they include at least this change https://github.com/SUSE-Cloud/socok8s/pull/259

Anyway, we could base it on airship files if you think it's better

File: playbooks/roles/airship-setup-deployer/tasks/kubectl-install.yml (hunk 1 above)
Comment (jsuchome) [reply playbooks/roles/airship-setup-deployer/tasks/kubectl-install.yml]: Right, it was changed with https://github.com/SUSE-Cloud/socok8s/pull/259

File: playbooks/roles/deploy-osh/tasks/helm-run.yml [hunk 6]
```diff
@@ -1,12 +1,14 @@
 ---
 # Inspired by https://github.com/openstack/openstack-helm-infra/blob/master/roles/build-helm-packages/tasks/setup-helm-serve.yaml
 - name: Copy systemd unit into place for helm server if it is not present
+  become: yes
```
Comment (Itxaka) [inline playbooks/roles/deploy-osh/tasks/helm-run.yml]: is this really needed? We are already using the root user for deployment so it should not be necessary to do this unless Im missing something?

Comment (jsuchome) [reply playbooks/roles/deploy-osh/tasks/helm-run.yml]: I am not sure, I was also surprised.
I was just syncing the code with current master and those changes were introduced with https://github.com/SUSE-Cloud/socok8s/pull/273 without specific explanation

**Comment by Itxaka** [comment]
duplicating my comment here from the chat:

why not create a simple role for client apps that include helm and kubectl + maybe other optional apps like kubebox[0], kube-ops-view[1] or kubectx[2]


[0] https://github.com/astefanutti/kubebox
[1] https://github.com/hjacobs/kube-ops-view
[2] https://github.com/ahmetb/kubectx

File: playbooks/roles/deploy-osh/tasks/helm-run.yml (hunk 6 above)
Comment (Itxaka) [reply playbooks/roles/deploy-osh/tasks/helm-run.yml]: yikes

**Comment by jsuchome** [comment]
@Itxaka sure, but that's out of scope of this PR. Actually I think Jira ticket for such work should be created

File: playbooks/roles/deploy-osh/tasks/helm-run.yml (hunk 6 above)
Comment (jgu17) [reply playbooks/roles/deploy-osh/tasks/helm-run.yml]: when we run the ansible playbooks on deployer, we should be able to run not as a root user, which is a security recommendation/practice we've been following.

**Comment by jgu17** [comment]
> @jgu17 I removed the cloud config file change. Still keeping OSH files as the only one, as they include at least this change #259
> Anyway, we could base it on airship files if you think it's better

@jsuchome I still prefer to base the file in the airship-setup-deployer role for two reasons: 1) the airship-setup-deployer is more specific/modular in what it does than deploy_osh, and it was created with the purpose for deploy-osh nd deploy-airship toles to share and reuse code in mind; (2) deploy_osh has a lot of things in it fpr helm developers and have deploy_airship to reference that seems not as clean.

**Comment by jsuchome** [comment]
@jgu17 ok, I will rework it so it is the other way

**Comment by evrardjp** [comment]
I would just put things into a single role which can be reused for both parts.

**Comment by jgu17** [comment]
> I would just put things into a single role which can be reused for both parts.

Agreed with JP here. airship_setup_deployer is set for this purpose (we can rename it to reflect that is a shared role by airship and osh).

**Comment by jsuchome** [comment]
Closing in favor of https://github.com/SUSE-Cloud/socok8s/pull/281

**Pull request closed without merging**